coral-common/src/main/java/com/linkedin/coral/common/TypeConverter.java

```diff
@@ -138,12 +138,17 @@ public static RelDataType convert(StructTypeInfo structType, final RelDataTypeFa
 
   // Mimic the StructTypeInfo conversion to convert a UnionTypeInfo to the corresponding RelDataType
   // The schema of output Struct conforms to https://github.com/trinodb/trino/pull/3483
+  // except we adopted "integer" for the type of "tag" field instead of "tinyint" in the Trino patch
+  // for compatibility with other platforms that Iceberg at currently doesn't support tinyint type.
```

Comment:
nit: `that Iceberg currently doesn't support tinyint type.`?

Reply:
Addressed, thanks for catching